pkg/controller/job/job_controller.go

```diff
@@ -705,6 +705,14 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	action := metrics.JobSyncActionReconciling
 
 	defer func() {
+		// Let us queue the job back accounting to the activeDeadlineSeconds. The duration after which it should be
+		// re-queued is activeDeadlineSeconds - time spent after start time.
+		if !jobSuspended(&job) && job.Spec.ActiveDeadlineSeconds != nil {
```

Comment:
Why do this in defer? It's harder to read and doesn't add much value. You can just do it after the conditional in lines 759 to 762.

Reply:
Ack. Changes have been addressed.

Reply:
The original intent of keeping in defer was to account for any other return from the `syncJob` other than job finished cases.